### PR TITLE
Test: Fix PHP 8.1 issues with `ilAssQuestionHintList`

### PR DESCRIPTION
**Pull request by mjansenDatabay**

### CODE REVIEW

File: Modules/TestQuestionPool/classes/class.ilAssQuestionHintList.php [hunk 1]
```diff
@@ -19,102 +19,53 @@
...
-     * iterator interface method
-     *
-     * @access	public
-     * @return	mixed
-     */
-    public function next()
+    public function next(): void
     {
-        return next($this->questionHints);
+        next($this->questionHints);
```
Comment (mbecker-databay) [inline Modules/TestQuestionPool/classes/class.ilAssQuestionHintList.php:45]: Is this correct? I'd think next is supposed to return a value of mixed

File: Modules/TestQuestionPool/classes/class.ilAssQuestionHintList.php [hunk 2]
```diff
@@ -19,102 +19,53 @@
...
-    /**
-     * iterator interface method
-     *
-     * @access	public
-     * @return	mixed
-     */
-    public function key()
+    public function key(): int
     {
         return key($this->questionHints);
```
Comment (mbecker-databay) [inline Modules/TestQuestionPool/classes/class.ilAssQuestionHintList.php:50]: Is this correct? Per docs, key returns int|string|null

File: Modules/TestQuestionPool/classes/class.ilAssQuestionHintList.php (hunk 1 above)
Comment (mjansenDatabay) [reply Modules/TestQuestionPool/classes/class.ilAssQuestionHintList.php:45]: According to the docs, next returns nothing

File: Modules/TestQuestionPool/classes/class.ilAssQuestionHintList.php (hunk 2 above)
Comment (mjansenDatabay) [reply Modules/TestQuestionPool/classes/class.ilAssQuestionHintList.php:50]: You can add a precise type in derivatives

**Pull request merged**